src/main/java/duke/DateTimeParser.java

```diff
@@ -0,0 +1,127 @@
+package duke;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+
+/**
+ * Contains functionality for parsing date and time input, as well as formatting
+ * LocalDate and LocalDateTime objects.
+ */
+public class DateTimeParser {
+
+    private static DateTimeFormatter parser = new DateTimeFormatterBuilder()
```

Comment:
I was impressed by your DateTimeParser (did not know of these methods I could use). Your use of appendOptionals to give the user more options to input the date has also inspired me to do better! However, Maybe you could have added a short comment here to explain the parser member as there does seem to be some implicit logic involved?